extending/plugins.rst

```diff
@@ -0,0 +1,134 @@
+.. _plugins:
+
+Plugins [EXPERIMENTAL]
+======================
+
+The Conan plugins is a feature intended to extend the Conan functionalities and let users customize the client behavior at determined
+points.
+
+Plugin structure
+----------------
+
+Plugins are Python files containing **pre** and **post** functions that will be executed prior and after a determined task performed by the
+Conan client. Those tasks could be Conan commands, recipe interactions such as exporting or packaging or interactions with the remotes.
+
+Here you can see an example of a simple plugin:
+
+.. code-block:: python
+   :caption: *example_plugin.py*
+
+    from conans import tools
+
+
+    def pre_export(output, conanfile, conanfile_path, reference, **kwargs):
+        conanfile_content = tools.load(conanfile_path)
+        test = "[RECIPE METADATA]"
```

Comment:
Use the recipe reference instead of (or added to) this hardcoded string:

```python
test = str(reference)  #  Would resolve to 'name/version@user/channel'
```

or do something with the `reference.name`, `reference.version`,...